File: net/management-api-v2/cm_api_v2_post_asset_folders.cs
// Tip: Find more about .NET SDKs at https://kontent.ai/learn/net
using Kontent.Ai.Management;
using Kontent.Ai.Management.Configuration;
using Kontent.Ai.Management.Models.AssetFolders;

// Or register it through DI with services.AddManagementClient()
using var client = new ManagementClient(new ManagementOptions
{
    ApiKey = "KONTENT_AI_MANAGEMENT_API_KEY",
    EnvironmentId = "KONTENT_AI_ENVIRONMENT_ID"
});

var response = (await client.CreateAssetFoldersAsync(new AssetFolderCreateModel
{
    Folders =
    [
        new AssetFolderHierarchy
        {
            Name = "Top level folder",
            ExternalId = "top-folder",
            Codename = "top_folder",
            Folders =
            [
                new AssetFolderHierarchy
                {
                    Name = "Second level folder",
                    ExternalId = "second-folder",
                }
            ]
        }
    ]
})).EnsureSuccess();

File: ts/management-api-v2/cm_api_v2_post_asset_folders.ts
// Tip: Find more about JS/TS SDKs at https://kontent.ai/learn/javascript
import { ManagementClient } from '@kontent-ai/management-sdk';

const client = new ManagementClient({
  environmentId: 'KONTENT_AI_ENVIRONMENT_ID',
  apiKey: 'KONTENT_AI_MANAGEMENT_API_KEY'
});

const response = await client.addAssetFolders()
  .withData(
    {
      folders: [
        {
          id: '958001d8-2228-4373-b966-5262b5b96f71',
          name: 'Downloads',
          codename: 'downloads',
          external_id: 'folder-with-downloadable-assets',
          folders: [
            {
              id: '9ca927b6-6e4d-4d6b-81e3-ec5e8f7772a0',
              name: 'Archives',
              codename: 'archives',
              external_id: 'folder-with-downloadable-archives'
            }
          ]
        },
        {
          id: '9ca927b6-6e4d-4d6b-81e3-ec5e8f7772a0',
          name: 'Legal documents',
          external_id: 'folder-documents'
        }
      ]
    }
  )
  .toPromise();